bench/Chainweb/Pact/Backend/Bench.hs

```diff
@@ -372,3 +375,34 @@ cpBenchSampleKeys numSampleEvents cp =
                     writeRow db Update ut f rowkeya b
                     writeRow db Update ut f rowkeyb a
         void $ _cpSave _cpeCheckpointer hash02
+
+
+cpBenchLookupProcessedTx :: Int -> CheckpointEnv -> C.Benchmark
+cpBenchLookupProcessedTx transactionCount cp = C.env (setup' cp) $ \ ~(ut) ->
+    C.bench benchname $ C.nfIO (go cp ut)
+  where
+    benchname = "lookupProcessedTx/transactionCount=" ++ show transactionCount
+    transaction (NoopNFData ut) db = incIntegerAtKey db ut f k 1
+    setup' CheckpointEnv{..} = do
+        usertablename <- _cpRestore _cpeCheckpointer Nothing >>= \case
+            PactDbEnv' db ->
+              setupUserTable db $ \ut -> writeRow db Insert ut f k 1
+        _cpSave _cpeCheckpointer hash01
+
+        _cpRestore _cpeCheckpointer (Just (BlockHeight 1, hash01)) >>= \case
+            PactDbEnv' pactdbenv ->
+              replicateM_ transactionCount (transaction usertablename pactdbenv)
+        void $ _cpSave _cpeCheckpointer hash02
+
+        return usertablename
+
+    f = "f"
+    k = "k"
+
+    hash01 = BlockHash $ unsafeMerkleLogHash "0000000000000000000000000000001a"
+    hash02 = BlockHash $ unsafeMerkleLogHash "0000000000000000000000000000002a"
+
+    go CheckpointEnv{..} (NoopNFData ut) = do
+        _cpRestore _cpeCheckpointer (Just (BlockHeight 2, hash02)) >>= \case
+          PactDbEnv' db ->
+            _cpLookupProcessedTx _cpeCheckpointer Nothing (V.fromList [Pact.TypedHash "" | _ <- [1..transactionCount]])
```

Comment:
@edmundnoble I added a simple benchmark where I just pass empty txhashes because I don't know how to calculate them on this level. There are request keys but we operate on much lower level here in the `setup'`.

Here are the results:

```
benchmarking pact-backend/batchedCheckpointer/lookupProcessedTx/transactionCount=100
time                 157.0 μs   (149.1 μs .. 166.8 μs)
                     0.988 R²   (0.976 R² .. 0.999 R²)
mean                 149.4 μs   (147.8 μs .. 153.9 μs)
std dev              8.776 μs   (4.321 μs .. 16.49 μs)
variance introduced by outliers: 58% (severely inflated)

benchmarking pact-backend/batchedCheckpointer/lookupProcessedTx/transactionCount=1000
time                 914.2 μs   (900.4 μs .. 933.6 μs)
                     0.998 R²   (0.996 R² .. 0.999 R²)
mean                 912.3 μs   (905.7 μs .. 921.5 μs)
std dev              26.52 μs   (20.01 μs .. 37.84 μs)
variance introduced by outliers: 18% (moderately inflated)
```

And these results are without batching with the following implementation:

```
void $ mapM (\i -> _cpLookupProcessedTx _cpeCheckpointer Nothing (V.fromList [i])) [Pact.TypedHash "" | _ <- [1..transactionCount]]
```

```
benchmarking pact-backend/batchedCheckpointer/lookupProcessedTx/transactionCount=100
time                 2.294 ms   (2.205 ms .. 2.406 ms)
                     0.983 R²   (0.973 R² .. 0.992 R²)
mean                 2.309 ms   (2.261 ms .. 2.372 ms)
std dev              199.3 μs   (159.0 μs .. 249.0 μs)
variance introduced by outliers: 60% (severely inflated)

benchmarking pact-backend/batchedCheckpointer/lookupProcessedTx/transactionCount=1000
time                 29.24 ms   (24.26 ms .. 31.75 ms)
                     0.948 R²   (0.911 R² .. 0.981 R²)
mean                 24.60 ms   (23.48 ms .. 26.24 ms)
std dev              3.407 ms   (2.490 ms .. 4.185 ms)
variance introduced by outliers: 61% (severely inflated)
```

Reply:
Looks great.